src/cli/command-processor.ts

```diff
@@ -0,0 +1,242 @@
+        demandOption: false,
+      })
+      .option('params', {
+        describe: 'The params argument for the execute command, in JSON format',
+        type: 'string',
+        demandOption: false,
+      })
+      .option('custom_data', {
+        describe:
+          'The customData argument for the execute command, in JSON format',
+        type: 'string',
+        demandOption: false,
+      })
+      .parse(userCommand, {}, (error, argv) => {
+        if (error !== null) {
+          throw error;
+        }
+        return argv;
+      });
+
+    /**
+     * Validate arguments and return an IntentMessage
+     */
+    switch (argv.intent_type) {
+      case 'IDENTIFY':
+        if (argv.discovery_buffer === undefined) {
+          throw new Error(
+            'discovery_buffer is required to trigger an Identify intent'
+          );
+        }
+        if (argv.device_id === undefined) {
+          throw new Error(
+            'device_id is required to trigger an Identify intent'
+          );
+        }
+        return new IdentifyMessage(
+          argv.request_id,
+          argv.discovery_buffer,
+          argv.device_id
+        );
+      case 'EXECUTE':
+        if (argv.local_device_id === undefined) {
+          throw new Error(
+            'local_device_id is required to trigger an Execute intent'
+          );
+        }
+        if (argv.command === undefined) {
+          throw new Error('command is required to trigger an Execute intent');
+        }
+        if (argv.params === undefined) {
+          throw new Error('paramns is required to trigger an Execute intent');
+        }
+        if (argv.custom_data === undefined) {
+          throw new Error(
+            'custom_data is required to trigger an Execute intent'
+          );
+        }
+        return new ExecuteMessage(
+          argv.request_id,
+          argv.local_device_id,
+          argv.command,
+          JSON.parse(argv.params),
+          JSON.parse(argv.custom_data)
+        );
+      default:
+        throw new Error('Unsupported value for intent_type argument.');
+    }
+  }
+
+  /**
+   * The main user input to response loop.
+   * Recieves raw user input and attempts to parse and forward
+   * messages to the worker thread.
+   */
+  async processUserInput(): Promise<void> {
+    // Open a readline interface
+    const readlineInterface = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    // Flag for signaling app exit.
+    let exit = false;
+    while (!exit) {
+      // Wrap the asyncronous question/response sequence in a promise to delay execution.
+      await new Promise(resolve => {
+        readlineInterface.question('Awaiting input...\n', async input => {
+          if (input.length === 0) {
+            resolve();
+            return;
+          }
+
+          //Parse the command.
+          const command = input.split(' ')[0];
+
+          if (input === 'exit') {
```

Comment:
May want to also include the `.exit` string used by the Node terminal environment.

Reply:
I moved the `exit` command to yargs.